### ISSUE TITLE
Bad visualization if there are many levels

### ISSUE THREAD

**Issue by frederikziebell** [csoneson/ExploreModelMatrix#5]
When there are many levels for some of the covariates in the design, text labels overlap and nothing can be read. Maybe one could switch then to a some sort of co-occurence heatmap?

```
library("pheatmap")
library("RColorBrewer")


set.seed(1)

n_samples <- 1000
n_conditions <- 50
n_batches <- 10

design <- data.frame(
  sample = paste0("sample_",1:n_samples),
  condition = paste0("condition_",rep_len(1:n_conditions,n_samples)),
  batch = sample(paste0("batch_",rep_len(1:n_batches,n_samples))),
  stringsAsFactors = FALSE
)

# co-occurence matrix
mat <- rep(0,n_conditions*n_batches) %>%
  matrix(ncol=n_conditions) %>%
  `colnames<-`(unique(design$condition)) %>%
  `rownames<-`(unique(design$batch))

mat <- reduce(1:nrow(design), 
       function(x, y) {
         x[design[[y,3]], design[[y,2]]] <- x[design[[y,3]], design[[y,2]]] + 1
         x
        }, .init=mat)
  

pheatmap(mat, color = colorRampPalette(brewer.pal(n = 7, name ="Blues"))(100))
```

I personally like such heatmaps, because they give an idea of how balanced an experimental design is, if there will be batches and one wants to randomize appropriately.

**Comment by csoneson** [comment]
Thanks. I think such a heatmap can be informative as an additional plot. However, it doesn't replace the current plot, whose main purpose is to visualize the value of the linear predictor for each combination of predictor values. Note that you can change the size of the current plot, as well as the size of the text (and flip the x/y axes), which will allow you to get a figure where the text doesn't overlap between fields.

**Comment by csoneson** [comment]
This has been included in v0.2.0. Thanks for the suggestion.
